### PR TITLE
Feature/add proxy settings

### LINKED ISSUE #122: Up to date random UA via real world browser usage statistic
I think the list of user agents is too small ([USER_AGENT_LIST](https://github.com/bisoncorps/search-engine-parser/blob/master/search_engine_parser/core/utils.py#L10)).

In other projects, I use (successfully) this library: [fake-user agent](https://github.com/hellysmile/fake-useragent). Up to date simple useragent faker with real world database.

In `utils.py`, `get_rand_user_agent()` maybe like this :
```
from fake_useragent import UserAgent

def get_rand_user_agent():
   ua = UserAgent()
   # Return random UA via real world browser usage statistic
   return ua.random
```

### PR DESCRIPTION
**Pull request by deven96**
Fix #122 and #119

### CODE REVIEW

File: search_engine_parser/core/utils.py [hunk 1]
```diff
@@ -21,7 +22,13 @@
 
 
 def get_rand_user_agent():
-    return random.choice(USER_AGENT_LIST)
+    user_agent = random.choice(USER_AGENT_LIST)
+    try:
+        user_agent = UserAgent().random
+    except:
+       pass
```
Comment (MeNsaaH) [inline search_engine_parser/core/utils.py:29]: Does the library throw any exceptions? We could do 
```python
  user_agent = UserAgent().random
  If not user_agent:
    user_agent = random.choice(USER_AGENT_LIST)
```

File: search_engine_parser/core/utils.py [hunk 2]
```diff
@@ -53,8 +69,13 @@ async def get_source(self, engine, url, headers, cache=True):
         if os.path.exists(cache_path) and cache:
             with open(cache_path, 'rb') as stream:
                 return pickle.load(stream), True
+        get_vars = { 'url':url, 'headers':headers }
+        if proxy and proxy_auth:
```
Comment (MeNsaaH) [inline search_engine_parser/core/utils.py:73]: For randomness sakes,  I don't know if we should make proxy apply sometimes and ignored sometimes.  Or in use cases where proxies should be used if the search hits a captcha

Comment (deven96) [reply search_engine_parser/core/utils.py:73]: If proxy isn't sent it isn't used. I don't think it's a good idea to provide the API and underneath make it random. If you don't want proxy, don't use proxy else use it

Comment (MeNsaaH) [reply search_engine_parser/core/utils.py:73]: Great

File: search_engine_parser/core/utils.py (hunk 1 above)
Comment (MeNsaaH) [reply search_engine_parser/core/utils.py:29]: ??

Comment (deven96) [reply search_engine_parser/core/utils.py:29]: Yeah it throws exceptions... hence the try except

**Pull request merged**